lldb/include/lldb/API/SBDefines.h

```diff
@@ -138,7 +138,10 @@ class LLDB_API SBUnixSignals;
 typedef bool (*SBBreakpointHitCallback)(void *baton, lldb::SBProcess &process,
                                         lldb::SBThread &thread,
                                         lldb::SBBreakpointLocation &location);
-
+typedef void (*SBNotificationCallback)(lldb::NotificationType type,
+                                       lldb::SBDebugger &,
+                                       lldb::SBExecutionContext &exe_ctx,
```

Comment:
Can we add accessors to the SBExecutionContext for accessing a debugger? We can modify the `lldb_private::ExecutionContext` to add a `lldb::DebuggerSP` and add accessors to the debugger object and then add that to the `lldb::SBExecutionContext" API?

Reply:
It's always been a little weird that ExecutionContext didn't have a debugger - though TTTT that only mattered for default constructed ones: so long as any of the other entities are filled in you can get to the debugger.
OTOH, that's a pretty big patch so if we want to go that way, that should be a separate patch.
Another way to do this would be to add an SBExecutionContext::GetDebugger and manage that all on the SB side?  After all, we haven't actually needed a Debugger in the ExecutionContext that I can remember, and this would be a much more limited change.
I'm also fine with passing the debugger alongside the SBExecutionContext, we do that in some other places.